keel-core/src/main/kotlin/com/netflix/spinnaker/keel/enforcers/EnvironmentExclusionEnforcer.kt

```diff
@@ -110,12 +114,13 @@ class EnvironmentExclusionEnforcer(
   }
 
   /**
-   * @throws EnvironmentCurrentlyBeingActedOn if there's an active deployment
+   * @throws ActiveDeployments  if there's an active deployment
    */
   private fun ensureNoActiveDeployments(deliveryConfig: DeliveryConfig, environment: Environment) {
-    /**
-     * To be implemented in a future PR.
-     */
+    artifactRepository.isDeployingTo(deliveryConfig, environment.name)
+    if(artifactRepository.isDeployingTo(deliveryConfig, environment.name)) {
+      throw ActiveDeployments(deliveryConfig, environment)
+    }
```

Comment:
This feels like using exceptions for control flow. Is this really an exception?

Reply:
I think this gets at a deep philosophical question. Here's my argument for using an exception here.

While "a deployment is active" is not exceptional from the perspective of keel as a whole, I think it is an exceptional thing to try do a verification while a deployment is happening. It's not a combination of events we typically expect to happen. I

I think that the "happy path" is that the enforcer's methods (`withVerificationLease`, `withActuationLease`) succeed.

Aesthetically, I also prefer the code structure when throwing the exception because then we can pass the exception -related info up the stack to the caller in a nice way.

Note, though, that I was a Python programmer in a previous life, where exceptions for control flow are so common that they're used for terminating loops! On the other hand, before that I was a C++ programmer, where using exceptions for control flow is severley frowned upon.